app/controllers/fhir/r4/api_controller.rb

```diff
@@ -457,6 +459,27 @@ def current_client_application
     OauthApplication.find_by(id: doorkeeper_token.application_id) if doorkeeper_token.application_id.present?
   end
 
+  # Determine if the patient's jurisdiction is valid for the requesting application
+  def jurisdiction_valid_for_client?(patient)
+    allowed_jurisdiction_ids = current_client_application.jurisdiction&.subtree&.pluck(:id) ||
```

Comment:
I'm having some trouble getting this to work with the user workflow. If I make a POST with a new patient with the jurisdiction `USA, State 1`, I get an error that the client does not have a jurisdiction. But the user who has granted permission to the client does have jurisdiction `USA, State 1`. And in fact I can see this if I change this line to be:
```
allowed_jurisdiction_ids = current_client_application.jurisdiction&.subtree&.pluck(:id) ||
                               current_client_application.user&.jurisdiction&.subtree&.pluck(:id) || 
                               [current_resource_owner.jurisdiction_id]
```
with the notable change being the last line, grabbing the `jurisdiction_id` from the `current_resource` owner. With this addition, POSTing a patient with jurisdiction `USA, State 1` succeeds, but if I change the jurisdiction to `USA, State 2`, I get the "Jurisdiction must be within the client application's jursdiction", as I would expect.

Using the `current_resource_owner` looks similar to me to the pattern @kylietmo used below in `accessible_patients`, so I'm wondering if `current_resource_owner` should be used to make this work for the user flow. Also possible this is just some issue with the way I have my client setup, but it seems reasonable that the `current_resource_owner` jurisdiction would be used, since they granted access to the client app.

Reply:
Yes, `current_resource_owner` should be used if present for the user workflow. Then you can just use the `viewable_patients` method on the `current_resource_owner`.

Reply:
Sorry made a bad assumption here. Good catch! This should be fixed now with the latest commit but please verify.